homeassistant/components/smarttub/climate.py

```diff
@@ -0,0 +1,116 @@
+"""Platform for climate integration."""
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_HEAT,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.util.temperature import convert as convert_temperature
+
+from . import SmartTubEntity
+from .const import DOMAIN, SMARTTUB_CONTROLLER
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up climate entity for the thermostat in the tub."""
+
+    controller = hass.data[DOMAIN][entry.unique_id][SMARTTUB_CONTROLLER]
+
+    entities = [SmartTubThermostat(controller, spa_id) for spa_id in controller.spa_ids]
+    if not any(entity.available for entity in entities):
+        raise PlatformNotReady
+
+    async_add_entities(entities)
+
+    return True
+
+
+class SmartTubThermostat(SmartTubEntity, ClimateEntity):
+    """The target water temperature for the spa."""
+
+    def __init__(self, controller, spa_id):
+        """Initialize the entity."""
+        super().__init__(controller, spa_id, "thermostat")
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation."""
+        heater_status = self.controller.get_heater_status(self.spa_id)
+        if heater_status == "ON":
+            return CURRENT_HVAC_HEAT
+        if heater_status == "OFF":
+            return CURRENT_HVAC_IDLE
+        raise NotImplementedError(f"unknown heater status: {heater_status}")
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes."""
+        return [HVAC_MODE_HEAT]
+
+    @property
+    def hvac_mode(self):
+        """Return the current hvac mode.
+
+        SmartTub devices don't seem to have the option of disabling the heater,
+        so this is always HVAC_MODE_HEAT.
```

Comment:
Is there no way to completely turn it off?

Reply:
Not that I know of. The vendor app does not have this functionality either.